src/eth/consensus/mod.rs

```diff
@@ -821,25 +846,25 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_peer_address_from_string_valid() {
+    fn test_peer_address_from_string_valid_http() {
         let input = "http://127.0.0.1:3000;3777".to_string();
         let result = PeerAddress::from_string(input);
 
         assert!(result.is_ok());
         let peer_address = result.unwrap();
-        assert_eq!(peer_address.address, "http://127.0.0.1");
+        assert_eq!(peer_address.address, "127.0.0.1");
         assert_eq!(peer_address.jsonrpc_port, 3000);
         assert_eq!(peer_address.grpc_port, 3777);
     }
 
     #[test]
-    fn test_another_peer_address_from_string_valid() {
+    fn test_peer_address_from_string_valid_https() {
         let input = "https://127.0.0.1:3000;3777".to_string();
         let result = PeerAddress::from_string(input);
 
         assert!(result.is_ok());
         let peer_address = result.unwrap();
-        assert_eq!(peer_address.address, "https://127.0.0.1");
+        assert_eq!(peer_address.address, "127.0.0.1");
         assert_eq!(peer_address.jsonrpc_port, 3000);
         assert_eq!(peer_address.grpc_port, 3777);
     }
```

Comment:
where is the protocol recorded? it makes a difference if it is http and https depending on if it is or not a kubernete access...

Reply:
I reverted the test cases and the fix. Now the changes are only applied to local context address instead of kubernetes+local.